### PR TITLE
switch to forms for new issues

### PR DESCRIPTION
**Pull request by kcbabo**
Switch from templates to GitHub forms for new issues. This provides a nicer experience for people filing issues (improved layout and controls) while also making sure we gather required information in a consistent fashion across issues.

You can see the new setup in this example repo:
https://github.com/kcbabo/edge-issue-forms/issues/new/choose

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/BUG-ISSUE.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      label: Expected Behavior
+      placeholder: When I performed x, I expected y to happen
+    validations:
+      required: true
+  - type: textarea
+    id: additional-context
+    attributes:
+      label: Additional Context
+      placeholder: |
+        - Kubernetes version
```
Comment (sam-heilbron) [inline .github/ISSUE_TEMPLATE/BUG-ISSUE.yml:47]: What do you think about making the Kubernetes version a required field?

**Review by sam-heilbron — COMMENTED** [review]
LGTM! One small question about making Kube version required. Thanks for making this happen!

File: .github/ISSUE_TEMPLATE/BUG-ISSUE.yml (hunk 1 above)
Comment (kcbabo) [reply .github/ISSUE_TEMPLATE/BUG-ISSUE.yml]: Sure thing. I added a new dropdown at the top for Kube version.

**Review by sam-heilbron — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/config.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+blank_issues_enabled: false
+contact_links:
+  - name: Talk with Us on Slack
+    url: https://slack.solo.io
+    about: For a fast response, consider asking your question on the Solo slack.
+  - name: Solo.io Documentation
+    url: https://docs.solo.io/gloo-edge/latest/guides
+    about: Check out the Troubleshooting section of the Gloo Edge docs.
```
Comment (jenshu) [inline .github/ISSUE_TEMPLATE/config.yml:8]: nit: there doesn't seem to be a Troubleshooting section in the gloo edge docs (at least not named as such).  There is a Debugging page under Operations though

File: .github/ISSUE_TEMPLATE/config.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+blank_issues_enabled: false
+contact_links:
+  - name: Talk with Us on Slack
```
Comment (jenshu) [inline .github/ISSUE_TEMPLATE/config.yml:3]: super nit: capitalization
```suggestion
  - name: Talk with us on Slack
```

File: .github/ISSUE_TEMPLATE/config.yml (hunk 2 above)
Comment (kcbabo) [reply .github/ISSUE_TEMPLATE/config.yml:8]: Doh! Thanks for catching that. Will issue another PR with this fix and the other typo you spotted.

Comment (kcbabo) [reply .github/ISSUE_TEMPLATE/config.yml]: Just realized this hasn't been merged yet, so just added the commit to my existing PR branch.

**Review by jenshu — APPROVED** [review]

**Pull request merged**